src/lib/Pjax/PjaxResponseMainContentMapper.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace EzSystems\HybridPlatformUi\Pjax;
+
+use EzSystems\HybridPlatformUi\Components\App;
+use EzSystems\HybridPlatformUi\Components\MainContent;
+use EzSystems\HybridPlatformUi\Mapper\MainContentMapper;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Maps a PJAX Response to the MainContent. And App.
+ */
+class PjaxResponseMainContentMapper implements MainContentMapper
+{
+    /**
+     * @var \EzSystems\HybridPlatformUi\Components\App
+     */
+    private $app;
+
+    public function __construct(App $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Response $response
+     *
+     * @throws \Exception
+     */
+    public function map($response)
+    {
+        if (!$response instanceof Response) {
+            throw new \InvalidArgumentException('Expected a \Symfony\Component\HttpFoundation\Response');
+        }
+
+        $responseContent = $response->getContent();
+
+        $errorHandling = libxml_use_internal_errors(true);
+
+        $doc = new \DOMDocument();
+        if (!$doc->loadHTML($responseContent)) {
+            $errors = libxml_get_errors();
+            libxml_use_internal_errors($errorHandling);
+            throw new \Exception(
+                "Error(s) occurred parsing the PJAX response:\n" .
+                implode("\n", $errors)
+            );
+        }
+
+        $xpath = new \DOMXPath($doc);
+        $nodeList = $xpath->query('//div[@data-name="title"]');
+        $title = $nodeList[0]->nodeValue;
+        $contentNodeList = $xpath->query('//div[@data-name="html"]//section[contains(concat(" ", normalize-space(@class), " "), " ez-serverside-content ")]');
+
+        $contentNode = $contentNodeList[0];
+        $attributes = 'ez-view-serversideview';
+        foreach (explode(' ', $contentNode->getAttribute('class')) as $classAttribute) {
+            if ($classAttribute !== 'ez-view-serversideview') {
+                $attributes .= " $classAttribute";
+            }
+        }
+        $content =
```

Comment:
depending on th anwser to my previous comment, I think `$content` could contain the content of the `div[data-name="html"]` which would greatly simplify the code there